pkg/providers/snow/reconciler/reconciler.go

```diff
@@ -2,19 +2,38 @@ package reconciler
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
 	"github.com/aws/eks-anywhere/pkg/controller"
 )
 
-type Reconciler struct{}
+type Reconciler struct {
+	client client.Client
+}
 
-func New() *Reconciler {
-	return &Reconciler{}
+func New(client client.Client) *Reconciler {
+	return &Reconciler{
+		client: client,
+	}
 }
 
-func (v *Reconciler) Reconcile(ctx context.Context, log logr.Logger, cluster *anywherev1.Cluster) (controller.Result, error) {
+func (r *Reconciler) Reconcile(ctx context.Context, log logr.Logger, cluster *anywherev1.Cluster) (controller.Result, error) {
+	for _, ref := range cluster.MachineConfigRefs() {
+		machineConfig := &anywherev1.SnowMachineConfig{}
+		if err := r.client.Get(ctx,
+			types.NamespacedName{Namespace: cluster.Namespace, Name: ref.Name},
+			machineConfig); err != nil {
+			return controller.Result{}, err
+		}
+		if !machineConfig.Status.SpecValid {
+			return controller.Result{}, fmt.Errorf("SnowMachineConfig %s is invalid", ref.Name)
```

Comment:
I think we should try to include the status error in the message

However, I don't think we should return an error here. That will trigger another reconciliation loop. But it will continue failing forever until the machine config is not changed.

So maybe just log it?

Reply:
Good catch, the cluster reconciler has already watched change for SnowMachineConfig. We should ignore validation errors here as it cannot be fixed by requeue and just wait for another notification.

We may want to change the vSphere reconciler to match this behaviour